src/controller/group-controller.js

```diff
@@ -0,0 +1,151 @@
+            error.path = "database"
+            console.error(error);
+            next(error)
+        }
+    } 
+
+    //GET groups/:groupid 처리
+    getGroupById = async(req,res,next) => {
+        const groupId = Number(req.params.groupId);
+        try{
+            const group = await groupService.getGroupById(groupId);
+            if (!group){
+                let error = new Error;
+                error.statusCode = 404;
+                error.message = "group not found"
+                next(error);
+            }
+            return res.status(200).send(group);
+
+        }catch(error){
+            error.statusCode = 500;
+            error.message = "server Error(Database)"
+            error.path = "database"
+            console.error(error);
+            next(error);
+        }
+    }
+
+    //PATCH METHOD 처리
+    modifyGroup = async(req,res,next) => {
+        const groupId = Number(req.params.groupId);
+
+        const {name, description,
+                ownerNickname, ownerPassword, 
+                photoUrl, tags, goalRep, 
+                discordWebhookUrl, discordInviteUrl} = req.body
+
+        const data = {groupId,name, description,
+                ownerNickname, ownerPassword, 
+                photoUrl, tags, goalRep, 
+                discordWebhookUrl, discordInviteUrl}
+
+        const group = await groupService.getGroupById(groupId);
+
+        if (!group){
+            let error = new Error;
+            error.statusCode = 404;
+            error.message = "group not found"
+            next(error);
+        }
+
+        try{
+            const modifiedGroupAndTag = await groupService.modifyGroup(data);
+            
+            return res.status(200).send(modifiedGroupAndTag);
+        }catch(error){
+            res.send(error);
+            console.error(error);
+        }
+            
+    }
+    
+    //DELETE METHOD 처리
+    deleteGroup = async (req,res,next) => {
+        let inputPassword  = req.body;
+        let groupId = req.params.groupId;
+        groupId = Number(groupId);
+
+        const group = await groupRepository.GetGroupByIdAll(groupId);
+        if (!group){
+            let error = new Error;
+            error.statusCode = 404;
+            error.message = "Group not found"
+            next(error) ;
+        }
+
+        try{
+            const groupPassword = await groupRepository.GetPassword(groupId);
+        }catch(error){
+            next(error);
+        }
+        
+        const reqPassword = inputPassword.ownerPassword;
+
+        if (groupPassword == reqPassword){
+            await groupRepository.DeleteGroup(groupId);
+            console.log("비밀번호 인증 성공")
+            return res.status(200).send(groupPassword);
+        }else{
+            let error = new Error;
+            error.statusCode = 401;
+            error.message = "wrong password"
+            error.path = 'password'
+            next(error) ;
+        }
+    }
```

Comment:
```suggestion
    deleteGroup = async (req, res, next) => {
        try {
            const groupId = Number(req.params.groupId);
            const { ownerPassword } = req.body;
        
            await groupService.deleteGroup(groupId, ownerPassword); //  서비스에 로직 위임
        
            return res.status(200).json({ message: "Group successfully deleted" });
    } catch (error) {
        next(error);
    }
```
비밀번호 비교와 같은 비즈니스 로직은 서비스에서 처리하는 것이 좋을 것 같습니다!

Reply:
서비스에서 원래 처리하려고 했었는데 throw를 이용하여 error를 던져도, 에러 헨들러에서 못 받는 것 같아서 옮겨봤습니다! 조금 더 공부해보고 수정하겠습니다!